### ISSUE TITLE
Taskbar is not responsive in fullscreen programs (E.g MC or Paladins)

### ISSUE THREAD

**Issue by socksprox** [linuxmint/cinnamon#11658]
### Distribution

21.1

### Package version

5.6.8

### Graphics hardware in use

NVIDIA Zotac GeForce RTX 2070 Super Mini

### Frequency

Always

### Bug description

The taskbar and the start menu get issue, i have recorded some videos, here they are: 

Paladins: https://streamable.com/xobtsv
Minecraft (PrismMC): https://streamable.com/f5jp2e

### Steps to reproduce

- Open a fullscreen app (where taskbar is hidden). I only tried with those two mentioned apove
- Try open taskbar with windows key and click icons like the speaker to change volume

### Expected behavior

Icon is responsive and volume menu opens

**Comment by Secret-chest** [comment]
It is incredibly weird (with Minecraft at least). Focusing another window immediately minimises Minecraft. As a workaround I use the desktop's fullscreen mode (set to Alt+F11) and make sure the Minecraft fullscreen is not enabled.

**Comment by chocolateimage** [comment]
This issue **needs** to be fixed because I have the issue for a long time and it's very annoying.

**Comment by PalmTamino** [comment]
Still not fixed in 21.2. This is a deal breaker😕

**Comment by nkuttkat** [comment]
This happens to me whenever I'm playing a Steam game in full-screen. I can open the Menu and start any application but the taskbar itself is unresponsive or rather disappears

**Comment by PalmTamino** [comment]
Ridiculous that this is still a thing and stays unpatched

**Comment by NoobSam777** [comment]
> This happens to me whenever I'm playing a Steam game in full-screen. I can open the Menu and start any application but the taskbar itself is unresponsive or rather disappears

This is making me frustrated

**Comment by fredcw** [comment]
I don't think that this is a bug but rather a feature request because the panel was not designed to be available over a full screen window. The panel only shows because you opened the menu with the win key so as soon as the menu closes, the panel disappears. If the panel remained visible, you'd then need a way to hide it again.

However, I think it would be a great feature to be able access the panel during a full screen game.

**Comment by NoobSam777** [comment]
> I don't think that this is a bug but rather a feature request because the panel was not designed to be available over a full screen window. The panel only shows because you opened the menu with the win key so as soon as the menu closes, the panel disappears. If the panel remained visible, you'd then need a way to hide it again.
> 
> However, I think it would be a great feature to be able access the panel during a full screen game.

Nice point

**Comment by Konstantin749** [comment]
> I don't think that this is a bug but rather a feature request because the panel was not designed to be available over a full screen window. The panel only shows because you opened the menu with the win key so as soon as the menu closes, the panel disappears. If the panel remained visible, you'd then need a way to hide it again.
> 
> However, I think it would be a great feature to be able access the panel during a full screen game.

But this makes no sense. If panel is visible it should be usable. Also Linux Mint positions itself as "friendly operating system" for Windows users and taskbar works there if it is visible.

**Comment by iWisp360** [comment]
Still not fixed?

**Comment by fredcw** [comment]
A solution is to not use super to show the menu. Instead try these:

* super-d to show desktop
* alt-space followed by n to minimise window/game
* Set a keyboard shortcut to minimise window in keyboard settings (alt-F3 for instance is not used for anything else by default)

**Comment by bcol80** [comment]
Switched over from Windows a couple months ago to Mint and this behavior is really annoying. What's the point in even letting you even see the menu with the super key when using a fullscreen app if you can't interact with it? And the 'solution' above is just cumbersome and counter intuitive. I know Linux DEs don't just wanna blindly follow Windows but in this case they should.

**Comment by Curthaldor** [comment]
Yup. Still an issue.

**Comment by freshefisch** [comment]
How is this a thing?? Just switched to Mint from Windows 11 and loving it so far, but this is really really annoying :/

**Comment by SpeeQz1** [comment]
Can report that this behavior is still present on X11 for Mint 22.0.
What are your thoughts on this @mtwebster?

**Comment by coliumsic** [comment]
Running on Linux Mint 22.1 Cinnamon, still an issue.

**Comment by LinuxGuy-cyber** [comment]
**+1 UPVOTED**

This needs to be fixed!

**Comment by socksprox** [comment]
> **+1 UPVOTED**
> 
> This needs to be fixed!
Well, the milestone of 2 years has been reached. I switched away from Cinnamon because this issue is so damn annoying for me

**Comment by LinuxGuy-cyber** [comment]
> > **+1 UPVOTED**
> > This needs to be fixed!
> > Well, the milestone of 2 years has been reached. I switched away from Cinnamon because this issue is so damn annoying for me

Good for you, I guess.  But the other choices quite don't work for me.
GNOME: I'd rather stick with 1 caveat of Mint, rather than 1000s of GNOME.
XFCE, MATE, LXQt, etc. Lightweight distros: My laptop is **NOT** from the 90s.
KDE Plasma: Of course, I don't want to deal with fixing my system every few months. (I like KDE btw)
Cosmic: Would probably give it a try, but it's also similar to GNOME, although less annoying.
Tiling Window Managers: Sorry, maybe I'd stick to being a boomer rather than memorising each command to launch every program on my PC. **I'd never use Arch btw... :)**

Cinnamon was rightfully designed. Not a lot of features, not a scarcity of features. Perfect middle ground.

**Comment by Nullcaller** [comment]
To me, the problem isn't even the taskbar itself not working properly. It's about asking yourself "What do I even do, now that it doesn't work properly?" when you inevitably get affected by this bug. The only solution that immediately came to mind in my case, is to close the fullscreen application. Using shortcuts outlined here is a better solution, but they all have disadvantages.

* alt-space followed by n to minimise window/game
* Set a keyboard shortcut to minimise window in keyboard settings. Alt-F3 for instance is not used for anything else by default (although be careful to not press alt-F4 because that closes the window/game.)

These don't always work because some applications don't allow you to minimize them, like games. I've encountered this with R.E.P.O. run with Proton.

* super-d to show desktop

This, though it does work reliably, just nukes all windows from all screens, forcing you to alt-tab into all of the windows you want back on your screen(s).

* switch to another workspace (Ctrl + Alt + Left / Right Arrow)

This isn't a good solution because opening a window in another workspace isn't really what I want. It isn't intuitive in any way, and dragging the window back into the workspace I want it in feels like an unneeded extra step.

There's another half-solution, which is better, in my opinion:

* **Press super, and just start typing. The text input goes into search, as it does on Windows and other Linucies. Just type in the application name and press enter. And for everything else, there's ~~Mast-~~ alt-tab.**

I've been using Mint for a week now. I've previously been using Ubuntu on my laptop, gave Mint a go on the desktop. Cinnamon is great, stable software. But the sooner this bug gets fixed, the better.

**Comment by LinuxGuy-cyber** [comment]
> To me, the problem isn't even the taskbar itself not working properly. It's about asking yourself "What do I even do, now that it doesn't work properly?" when you inevitably get affected by this bug. The only solution that immediately came to mind in my case, is to close the fullscreen application. Using shortcuts outlined here is a better solution, but they all have severe disadvantages.
> 
>     * alt-space followed by n to minimise window/game
> 
>     * Set a keyboard shortcut to minimise window in keyboard settings. Alt-F3 for instance is not used for anything else by default (although be careful to not press alt-F4 because that closes the window/game.)
> 
> 
> These don't always work because some applications don't allow you to minimize them, like games. I've encountered this with R.E.P.O. run with Proton.
> 
>     * super-d to show desktop
> 
> 
> This, though it does work reliably, just nukes all windows from all screens, forcing you to alt-tab into all of the windows you want back on your screen(s).
> 
>     * switch to another workspace (Ctrl + Alt + Left / Right Arrow)
> 
> 
> This isn't a good solution because opening a window in another workspace isn't really what I want. It isn't intuitive in any way, and dragging the window back into the workspace I want it in feels like an unneeded extra step. Though it _is_ the best existing solution at this point in time.
> 
> There's another half-solution, which is better, in my opinion:
> 
>     * Press super, and just start typing. The text input goes into search, as it does on Windows and other Linucies. Just type in the application name and press enter. And for everything else, there's ~Mast-~ alt-tab.
> 
> 
> I've been using Mint for a week now. I've previously been using Ubuntu on my laptop, gave Mint a go on the desktop. Cinnamon is great, stable software. But the sooner this bug gets fixed, the better.

Well, afaik this isn't a "bug". It's the way Cinnamon was designed. But this is an awful **limitation**. It's been 2 years since this thread was opened, so there's very few chances of it even getting the attention of the devs now.

**Comment by claudiux** [comment]
When I set an application to full screen... You know what? It goes full screen! Any mouse action on the screen is intercepted by... my application, just as I wanted.

My keyboard shortcuts remain available (to change the sound volume or open the main menu, for example).

If I want to be able to access my panels, I configure my application in windowed mode. If this isn't possible, the culprit is this application.

**Comment by Nullcaller** [comment]
> When I set an application to full screen... You know what? It goes full screen! Any mouse action on the screen is intercepted by... my application, just as I wanted.

And when I press super, it brings up the taskbar, and any mouse action is intercepted by the taskbar, just as I wa- wait :)

> If I want to be able to access my panels, I configure my application in windowed mode. If this isn't possible, the culprit is this application.

Yeah, but the user doesn't really care. And don't shoot me, I'm just the messenger. The user expects the desktop environment to help them manage applications.

The OS itself, as a concept, exists for the same reason. It doesn't just go "okay, switch the processor into real mode and pass control" when you open an application. There's a balance, set mainly by user expectations.

**Comment by LinuxGuy-cyber** [comment]
Pulled this off Perplexity AI (Consequences of not having the feature):

![Image](https://github.com/user-attachments/assets/e3246624-9e48-4352-90f0-b2dfe68215a6)

**Comment by claudiux** [comment]
Yes, I am responsible for my choices. I don't need an AI to tell me that.

**Comment by LinuxGuy-cyber** [comment]
> Yes, I am responsible for my choices. I don't need an AI to tell me that.

Never explicitly mentioned you. No AI is telling you. I have highlighted the common problems (through) the AI.

(Not a native English speaker so not sure if you meant something else.)

**Comment by Nullcaller** [comment]
Not gonna lie, a bit weirded out by the use of AI here too. It just reiterates common-sense points in a distinctly LLM-ish way, you know, with every vanishingly miniscule yet linguistically differentiable aspect of a thing being outlined in a separate bullet point.

The point still stands, and the LLM did outline all issues correctly, but yeah, idk why it was needed.

About being responsible for your choices, though... Like, the point of this issue is not about responsibility. No one here thinks "Cinnamon devs, and @claudiux specifically, are responsible, and should feel bad about their actions". At least I don't, and in no way am I trying to imply that.

It's not about things being bad and someone being responsible. It's about things becoming better, more people loving the software you build and Linux in general. Because, as I've already said, it's great.

**Comment by claudiux** [comment]
"Yes, I am responsible for my choices." means: If I want an application to be full-screen, I accept that it reacts in full-screen mode.

_Don't talk to me about Windows. As Linus Torvalds says: "Computer is like air conditioning; it becomes useless when you open windows."_

**Comment by Nullcaller** [comment]
> _Don't talk to me about Windows._

Alright, let's talk Linux. With Ubuntu Desktop, you can press super, then click anything on your dock, and the click will not be redirected to the fullscreen application.

Or macOS. I haven't used macOS in a while, but I believe it works largely the same way in macOS.

> "Yes, I am responsible for my choices." means: If I want an application to be full-screen, I accept that it reacts in full-screen mode.

The point is, among users, there's a pretty common understanding of what full-screen is.

The concept of 'full-screen' is arbitrary and implementation-dependent, just like everything else that has to do with modern GUIs. It's a complete fabrication. It was never real. We made it all up.

But since most operating systems seem to have agreed on what full-screen means (and it doesn't mean that all mouse input is redirected to the application, no matter what), users have developed an understanding of how full-screen should work.

If you then go against that understanding, by making the desktop environment, idk, launch all of the world's nukes when a user tries to click on the taskbar while an application is in full-screen mode, the user isn't then magically made responsible for the genocide that follows, because "the user should've known what full-screen means, it means launching all of the world's nukes".

Because no, it has never been this way, nor could a user have reasonably expected it.

**Comment by fredcw** [comment]
I've looked into this. Unfortunately, it's not a trivial fix because of the way the code is structured.

**Comment by claudiux** [comment]
Super+D displays the Desktop. It works with a fullscreen app I use. Try with yours.

**Comment by Nullcaller** [comment]
Alright then. If the fix is very non-trivial, it's honestly fine. One last question though.

Is it possible, or, rather, is it reasonably doable, to make it so that you can use your keyboard to navigate the panels, once you press super? You know, press tab, and you can use the left/right arrow keys to select an applet, press tab again and you can peruse the system tray...

Just, you know, as a compromise. Better than super-d and workspace switching, less intuitive than using a mouse.

That's honestly the first thing I tried when I understood that mouse clicks don't work, but it seems that super just brings up the menu applet, and does nothing else.

Or is there already a way to do this? Maybe a hotkey that you can set in system settings?

**Comment by Nullcaller** [comment]
Oh, and also, to avoid user confusion, since it's obviously possible for Cinnamon to detect when an application is in full-screen mode and on the same screen as the taskbar, it'd pretty cool if Cinnamon displayed a little toast somewhere, saying something along the lines of "Tip: You can't click on the taskbar when an application is full-screen[, but you can use tab and arrow keys to navigate it*]". (* - if that's something that can reasonably be done) With an option to disable it in system settings, of course.

**Comment by claudiux** [comment]
> Alright then. If the fix is very non-trivial, it's honestly fine.

So, I close this issue. You also can change of keybindings in system settings.

For other feature requests, please open new ones.

**Comment by LinuxGuy-cyber** [comment]
Just asking, can this be considered adding during the port to Wayland?

**Comment by Soul128** [comment]
I am using Linux Mint 22.1. The taskbar is shown when I press Super in "windowed full-screen mode" in a game, but I can't click it or anything. When I click a window, the taskbar, or anything else, I am taken back into the game.
The only thing that works is the "Start" menu itself—for whatever reason, this menu appears in front of my full-screen game as expected , but the taskbar is not intractable.
It is also broken when the game is not in focus, and I press anything on the taskbar. I can see it but not click it.
Why is the taskbar even shown when it is not interactive? Why is it "click-through"?
Please don't show me things I can normally use/click, but in some random situations I’m not able to… that is very confusing.
